app/application/main.py

```diff
@@ -0,0 +1,50 @@
+import asyncio
+
+import logging
+import signal
+from datetime import datetime, timedelta
+
+from apscheduler.events import EVENT_JOB_ERROR, JobExecutionEvent
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+
+from apscheduler.triggers.interval import IntervalTrigger
+
+from logger.base import create_logger
+from logic.send_random_message import send_random_image_and_text
+
+
+def error_listener(event: JobExecutionEvent) -> None:
+    logging.getLogger('app').exception("Job raised an exception", exc_info=event.exception)
+
+
+if __name__ == '__main__':
+    logger = create_logger('app')
+    logging.getLogger('apscheduler.executors.default').disabled = True
+
+    loop = asyncio.new_event_loop()
+
+    scheduler = AsyncIOScheduler(event_loop=loop)
+    scheduler.add_listener(error_listener, EVENT_JOB_ERROR)
+
+    trigger = IntervalTrigger(minutes=60)
+    job = scheduler.add_job(send_random_image_and_text, trigger)
+
+    def graceful_stop():
```

Comment:
Try not to define your functions inside main code - it is considered  a bad practice

Reply:
Function extracted